### PR TITLE
feat(ui-context): Add selected theme to user context

### PR DESCRIPTION
**Pull request by Noha-ElAbrouki**
add theme to userContext defined to light or dark

### CODE REVIEW

File: packages/ui-context/src/types.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ export interface User {
   isExportButtonEnabled: boolean;
   locale: string;
   name: string;
+  theme: 'light' | 'dark';
```
Comment (Thebarda) [inline packages/ui-context/src/types.ts:9]: I would use an `enum` though.
Don't forget to export it in the `src/index.ts` file

File: packages/ui-context/src/userAtom.ts [hunk 2]
```diff
@@ -4,6 +4,6 @@ import { defaultUser } from './defaults';
 
 import { User } from '.';
 
-const userAtom = atom<User>(defaultUser);
+const userAtom = atom<User>(defaultUser as User);
```
Comment (Thebarda) [inline packages/ui-context/src/userAtom.ts:7]:
```suggestion
const userAtom = atom<User>(defaultUser);
```

File: packages/ui-context/src/types.ts (hunk 1 above)
Comment (bdauria) [reply packages/ui-context/src/types.ts:9]: She actually did that but I told her to do it that way 🙈 
Can you confirm it'd work as as string by the time we receive either 'light' or 'dark' from the API?

Comment (Noha-ElAbrouki) [reply packages/ui-context/src/types.ts:9]: l'api retourne un string , c'est pourquoi je n'ai pas utilisé enum

Comment (Thebarda) [reply packages/ui-context/src/types.ts:9]: BTW ts.data.json manage the conversion between strings and enums

Comment (bdauria) [reply packages/ui-context/src/types.ts:9]: alright, @Noha-ElAbrouki you can change it back to an enum then :)

Comment (Noha-ElAbrouki) [reply packages/ui-context/src/types.ts:9]: okey

File: packages/ui-context/src/types.ts [hunk 3]
```diff
@@ -6,10 +6,16 @@ export interface User {
...
   locale: string;
   name: string;
+  theme: Theme;
   timezone: string;
   use_deprecated_pages: boolean;
 }
 
+export enum Theme {
+  light,
+  dark,
```
Comment (Thebarda) [inline packages/ui-context/src/types.ts:16]:
```suggestion
  light = 'light',
  dark = 'dark',
```

**Review by Thebarda — APPROVED** [review]

File: packages/centreon-ui/src/utils/useLocaleDateTimeFormat/index.test.tsx [hunk 4]
```diff
@@ -9,9 +9,10 @@ import { render, RenderResult } from '@testing-library/react';
 import { Provider } from 'jotai';
 import { useUpdateAtom } from 'jotai/utils';
 
-import { userAtom } from '@centreon/ui-context';
+import {  userAtom } from '@centreon/ui-context';
 
 import useLocaleDateTimeFormat from '.';
+import { Theme } from '@centreon/ui-context/src/types';
```
Comment (bdauria) [inline packages/centreon-ui/src/utils/useLocaleDateTimeFormat/index.test.tsx:15]: The theme should be imported locally via centreon-ui

File: packages/centreon-ui/src/utils/useLocaleDateTimeFormat/localeFallback.test.tsx [hunk 5]
```diff
@@ -12,6 +12,7 @@ import { Provider } from 'jotai';
 import { userAtom } from '@centreon/ui-context';
 
 import useLocaleDateTimeFormat from '.';
+import { Theme } from '@centreon/ui-context/src/types';
```
Comment (bdauria) [inline packages/centreon-ui/src/utils/useLocaleDateTimeFormat/localeFallback.test.tsx:15]:
```suggestion
```

File: packages/centreon-ui/src/utils/useLocaleDateTimeFormat/index.test.tsx [hunk 6]
```diff
@@ -9,9 +9,10 @@ import { render, RenderResult } from '@testing-library/react';
 import { Provider } from 'jotai';
 import { useUpdateAtom } from 'jotai/utils';
 
-import { userAtom } from '@centreon/ui-context';
+import {  userAtom } from '@centreon/ui-context';
```
Comment (bdauria) [inline packages/centreon-ui/src/utils/useLocaleDateTimeFormat/index.test.tsx:12]:
```suggestion
import {  userAtom, Theme } from '@centreon/ui-context';
```

File: packages/centreon-ui/src/utils/useLocaleDateTimeFormat/index.test.tsx (hunk 4 above)
Comment (bdauria) [inline packages/centreon-ui/src/utils/useLocaleDateTimeFormat/index.test.tsx:15]:
```suggestion
```

File: packages/centreon-ui/src/utils/useLocaleDateTimeFormat/localeFallback.test.tsx [hunk 7]
```diff
@@ -12,6 +12,7 @@ import { Provider } from 'jotai';
 import { userAtom } from '@centreon/ui-context';
```
Comment (bdauria) [inline packages/centreon-ui/src/utils/useLocaleDateTimeFormat/localeFallback.test.tsx:12]:
```suggestion
import { userAtom, Theme } from '@centreon/ui-context';
```

**Comment by bdauria** [comment]
@Noha-ElAbrouki the Theme should be exported from the index.ts file from ui-context.

I also suggest that we rename it into UserTheme so that it dosen't conflict with the existig theme that we have in centreon-ui

File: packages/centreon-ui/src/utils/useLocaleDateTimeFormat/localeFallback.test.tsx [hunk 8]
```diff
@@ -32,6 +32,7 @@ const TestComponent = (): JSX.Element => {
       name: 'admin',
       timezone: 'Europe/Paris',
       use_deprecated_pages: false,
+      theme:UserTheme.light
```
Comment (Thebarda) [inline packages/centreon-ui/src/utils/useLocaleDateTimeFormat/localeFallback.test.tsx:35]:
```suggestion
      theme: UserTheme.light
```

**Review by Thebarda — COMMENTED** [review]
@Noha-ElAbrouki You have an ESLint errors on the following files: 
- 'centreon-frontend/packages/centreon-ui/src/utils/useLocaleDateTimeFormat/index.test.tsx'
- 'centreon-frontend/packages/centreon-ui/src/utils/useLocaleDateTimeFormat/localeFallback.test.tsx'

**Review by bdauria — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Pull request merged**